nixos/modules/services/backup/borgmatic.nix

```diff
@@ -100,5 +100,10 @@ in
 
     # Workaround: https://github.com/NixOS/nixpkgs/issues/81138
     systemd.timers.borgmatic.wantedBy = [ "timers.target" ];
+
+    system.checks = singleton (pkgs.runCommandCC "config validation" { } ''
+        ${pkgs.borgmatic}/bin/borgmatic -c ${cfgfile} config validate
```

Comment:
`cfgfile` could be empty if using another type of format, see line L92-L97

Can `system.checks` inherit `systemd.services.borgmatic.serviceConfig.EnvironmentFile`? Because I save the `encryption_passphrase` in environment.

Reply:
> cfgfile could be empty if using another type of format, see line L92-L97

good point, thanks!

> Can system.checks inherit systemd.services.borgmatic.serviceConfig.EnvironmentFile? Because I save the encryption_passphrase in environment.

system.checks is executed at build time, which means that its not possible without loading the file to the nix store.
I'm not sure if a missing `encryption_passphrase` would fail the check though.
Do you think it'd make sense to add an option to disable the check like with [varnish](https://github.com/NixOS/nixpkgs/blob/c5101e457206dd437330d283d6626944e28794b3/nixos/modules/services/web-servers/varnish/default.nix#L16)?

Reply:
> I'm not sure if a missing encryption_passphrase would fail the check though.

It will fail with exit 1

```
# head -2 /etc/borgmatic.d/filebrowser.yaml
compression: zstd
encryption_passphrase: ${FILEBROWSER_BORG_PASSPHRASE}

# borgmatic config validate
/etc/borgmatic.d/filebrowser.yaml: Error parsing configuration file
Cannot find variable FILEBROWSER_BORG_PASSPHRASE in environment
```

> Do you think it'd make sense to add an option to disable the check like with varnish?

Sounds good